### PR TITLE
feat(source): support postgres cdc connector

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Refine the WITH properties of CDC source
- Supports the Postgres CDC connector. (`connector='postgres-cdc'`)
```
create materialized source employee (
  empid INTEGER,
  name STRING,
  mgrid INTEGER,
 PRIMARY KEY (empid)
) with (
 connector = 'postgres-cdc',
 hostname = '127.0.0.1',
 port = '5432',
 username = 'postgres',
 password = '123456',
 database.name = 'dev',
 schema.name = 'public',
 table.name = 'te',
 slot.name = 'employee'  // PG replication slot, different table should be given different name
) row format debezium_json;
```
- Change some properties of the MySQL CDC connector
```
create materialized source products ( id INT,
 name STRING,
 description STRING,
 PRIMARY KEY (id)
) with (
 connector = 'mysql-cdc',   // it is 'cdc' previously
 hostname = '127.0.0.1',
 port = '3306',
 username = 'root',
 password = '123456',
 database.name = 'mydb',
 table.name = 'products',
 server.id = '5454',   // A numeric ID of the database client, which must be unique across processes in the MySQL cluster.
) row format debezium_json;

```

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators

### Release note

1. Supports the Postgres CDC connector. (`connector='postgres-cdc'`)
2. Changes to the MySQL CDC connector
    - connector name changed to `mysql-cdc`
    - user needs to specify the `server.id` when creating a MySQL CDC source

## Refer to a related PR or issue link (optional)
#6732

### CODE REVIEW

File: src/source/src/connector_source.rs [hunk 1]
```diff
@@ -187,7 +187,7 @@ pub struct ConnectorSource {
 
 impl ConnectorSource {
     #[allow(clippy::too_many_arguments)]
-    pub async fn new(
+    pub async fn new_v2(
```
[the remark this answers is not in the document]
Comment (StrikeW) [reply src/source/src/connector_source.rs:190]: It is only used in `SourceDescBuilderV2`, to make it more explicit.

File: e2e_test/source/cdc/cdc.load.slt [hunk 2]
```diff
@@ -75,11 +79,12 @@ create materialized source mytable (
   v3 STRING,
  PRIMARY KEY (v1)
 ) with (
- connector = 'cdc',
- database.hostname = 'mysql',
- database.port = '3306',
- database.user = 'root',
- database.password = '123456',
+ connector = 'mysql-cdc',
```
[the remark this answers is not in the document]
Comment (StrikeW) [reply e2e_test/source/cdc/cdc.load.slt:82]: It is unnecessary to use two fields. This style conforms to Flink CDC, I think it is easy to understand.

**Review by tabVersion — APPROVED** [review]
generally LGTM

File: src/source/src/connector_source.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (tabVersion) [reply src/source/src/connector_source.rs:190]: there is no `new` method in `ConnectorSource`. It is strange that we only have v2 here

**Comment by StrikeW** [comment]
> > slot.name = 'employee'  // PG replication slot, different table should be given different name
> 
> Is the user aware of PG's replication slot when they set up the CDC? I wonder if it's manually configured or automatically set by PG.

I think Debeizum will create the replication slot for the captured table using the `slot.name` as a name.
IMO, CDC connectors provided by RisingWave are just a proxy for the Debezium connectors, which should be explicitly documented in the user guide. User may need to refer the doc of Debezium for more [information](https://debezium.io/documentation/reference/1.9/connectors/postgresql.html#postgresql-server-configuration) :

Debezium uses PostgreSQL’s logical decoding, which uses replication slots. Replication slots are guaranteed to retain all WAL segments required for Debezium even during Debezium outages. For this reason, it is important to closely monitor replication slots to avoid too much disk consumption and other conditions that can happen such as catalog bloat if a replication slot stays unused for too long. For more information, see the [PostgreSQL streaming replication documentation](https://www.postgresql.org/docs/current/warm-standby.html#STREAMING-REPLICATION-SLOTS).

> > server.id = '5454',   // A numeric ID of the database client, which must be unique across processes in the MySQL cluster.
> 
> What if the server crashes and restarts?

When crash or restart, Meta will use the same properties to rebuild the connector source, that is the `server.id` will remain unchanged. If it becomes duplicated after the restart, user may need to drop the source and recreate it with a new value for `server.id` in current stage.

File: src/source/src/connector_source.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (StrikeW) [reply src/source/src/connector_source.rs:190]: Ok, I will revert it back.

**Pull request merged**